lib/Demangling/Demangler.cpp

```diff
@@ -139,6 +139,11 @@ bool swift::Demangle::isSwiftSymbol(llvm::StringRef mangledName) {
   return getManglingPrefixLength(mangledName) != 0;
 }
 
+bool swift::Demangle::isSwiftSymbol(const char *mangledName) {
+  StringRef mangledNameRef(mangledName, 4);
```

Comment:
What happens if a mangled name is shorter than 4 characters? I happened to notice it and "fixed" it in https://github.com/apple/swift/pull/14249 but now that I see you introduced this behaviour so recently, I'm now wondering if that was a mistake...

Reply:
Thanks for fixing; I think it was sloppiness on my part when I re-introduced an old API that already had these issues.